### PR TITLE
sample solution for supplying attributes to content areas

### PR DESCRIPTION
**Pull request by andrewmcodes**
### Summary

Example solution for supplying attributes to content areas

Related Issues #285

### Other Information

Results of pairing with @joelhawksley

### CODE REVIEW

**Comment by andrewmcodes** [comment]
ping @jonspalmer

File: test/view_component/view_component_test.rb [hunk 1]
```diff
@@ -504,6 +504,22 @@ def test_render_single_item_from_collection
     assert_selector("p", text: "On sale", count: 1)
   end
 
+  def test_content_area_named_attributes
+    render_inline(CardComponent.new) do |component|
+      component.with(:header, classes: "card-header mt-4", data: {controller: "card-header-controller"}) { "Card Header" }
```
Comment (muan) [inline test/view_component/view_component_test.rb:509]:
```suggestion
      component.with(:header, classes: "card-header mt-4") { "Card Header" }
```

**Comment by jonspalmer** [comment]
I'm a little confused about how this implementation solves the original problem in #285. This doesn't look like it allows arguments to be passed into the content areas. Instead it's provide a way to decorate those areas with other static properties. I don't think that's the same thing.

**Comment by joelhawksley** [comment]
Thanks for doing this spike @andrewmcodes! We're going pursue a different approach to this change ❤️

**Comment by andrewmcodes** [comment]
@joelhawksley sounds good! Sorry this change slipped off my radar due to some personal events but I am excited to see what everyone came up with! Thanks for working on this with me!

**Pull request closed without merging**